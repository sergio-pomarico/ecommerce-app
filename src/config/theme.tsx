import React, {ReactNode} from 'react';
import {
  createTheme,
  useTheme as useReTheme,
  ThemeProvider as ReStyleThemeProvider,
} from '@shopify/restyle';
import {
  ImageStyle,
  TextStyle,
  ViewStyle,
  FlexStyle,
  Dimensions,
} from 'react-native';

type NamedStyles<T> = {
  [P in keyof T]: ViewStyle | TextStyle | ImageStyle | FlexStyle;
};

const {width} = Dimensions.get('screen');
export const aspectRatio = width / 375;

const palette = {
  black: '#000000',
  blue: '#7DCCEC',
  lightBlue: '#D3F2FF',
  danger: '#E16C71',
  grey: '#9A9A9D',
  darkGrey: '#868686',
  lightGrey: '#F5F5F8',
  mediumGrey: '#C4C4C4',
  indigo: '#5956E9',
  ligthIndigo: '#F4F6FA',
  red: '#FA4A0C',
  white: '#FFFFFF',
};

const theme = createTheme({
  colors: {
    background: palette.lightGrey,
    primary: palette.indigo,
    text: palette.black,
    white: palette.white,
    blue: palette.blue,
    grey: palette.grey,
    darkGrey: palette.darkGrey,
    error: palette.danger,
  },
  spacing: {
    xs: 8,
    s: 16,
    m: 24,
    l: 32,
    xl: 64,
  },
  borderRadii: {
    s: 4,
    m: 10,
    l: 25,
    xl: 64,
  },
  breakpoints: {},
  textVariants: {
    hero: {
      color: 'white',
      fontSize: 50,
      fontFamily: 'Raleway-ExtraBold',
      lineHeight: 50,
      textAlign: 'center',
    },
    h1: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 32,
      lineHeight: 36,
      textAlign: 'center',
    },
    h2: {
      fontFamily: 'Raleway-Bold',
      fontSize: 28,
      lineHeight: 32,
      textAlign: 'center',
    },
    h3: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 24,
      lineHeight: 24,
      textAlign: 'center',
    },
    product_card_title: {
      fontFamily: 'Raleway-Regular',
      fontSize: 16,
      fontWeight: '600',
      lineHeight: 18,
    },
    price: {
      fontFamily: 'Raleway-Regular',
      fontSize: 15,
      fontWeight: '600',
      lineHeight: 17,
    },
    button_large: {
      fontFamily: 'Raleway-Bold',
      fontSize: 20,
      lineHeight: 24,
      textAlign: 'center',
    },
    button_medium: {
      fontFamily: 'Raleway-Bold',
      fontSize: 16,
      lineHeight: 18,
      textAlign: 'center',
    },
    label: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 15,
      lineHeight: 18,
      textTransform: 'capitalize',
    },
  },
});

export const ThemeProvider = ({children}: {children: ReactNode}) => (
  <ReStyleThemeProvider {...{theme}}>{children}</ReStyleThemeProvider>
);

export type Theme = typeof theme;
export const useTheme = () => useReTheme<Theme>();

export const makeStyle =
  <T extends NamedStyles<T>>(styles: (theme: Theme) => T) =>
  () => {
    const curretTheme = useTheme();
    return styles(curretTheme);
  };

export default theme;
