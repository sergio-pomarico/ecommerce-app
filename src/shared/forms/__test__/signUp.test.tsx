import React from 'react';
import {ThemeProvider} from '@config/theme';
import {
  render,
  fireEvent,
  waitFor,
  RenderAPI,
} from '@testing-library/react-native';

import SignUpForm from '@forms/signup';

describe('SignUp test suite', () => {
  const onSubmitMock = jest.fn();
  const onPressLoginMock = jest.fn();
  const emailMock = 'email@email.com';
  const passwordMock = '123445';
  let signUpComponent: RenderAPI;

  beforeEach(() => {
    signUpComponent = render(
      <ThemeProvider>
        <SignUpForm onSubmit={onSubmitMock} onPressLogin={onPressLoginMock} />
      </ThemeProvider>,
    );
    jest.clearAllMocks();
  });

  test('check if the SignUp Form render correctly', () => {
    const {toJSON} = signUpComponent;
    expect(toJSON()).toMatchSnapshot();
  });

  test('check if SignUp Form is submitting correctly', async () => {
    const {getByTestId} = signUpComponent;
    const email = getByTestId('input_email');
    const password = getByTestId('input_password');
    const confirm_password = getByTestId('input_comfirm_password');
    const buton = getByTestId('btn_sign_up');

    fireEvent.changeText(email, emailMock);
    fireEvent.changeText(password, passwordMock);
    fireEvent.changeText(confirm_password, passwordMock);
    fireEvent.press(buton);

    await waitFor(() => {
      expect(onSubmitMock).toHaveBeenCalledWith({
        email: emailMock,
        password: passwordMock,
        passwordConfirmation: passwordMock,
      });
    });
  });

  test('check if Login Form submit button is disabled', () => {
    const {getByTestId} = signUpComponent;
    const buton = getByTestId('btn_sign_up');
    fireEvent.press(buton);
    expect(onSubmitMock).not.toBeCalled();
  });
});
