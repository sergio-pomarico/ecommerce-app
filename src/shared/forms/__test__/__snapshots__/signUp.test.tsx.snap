// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp test suite check if the SignUp Form render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "flex": 1,
        "padding": 32,
        "paddingVertical": 64,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 120,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#9A9A9D",
            "borderBottomWidth": 0.5,
            "marginBottom": 8,
            "paddingBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginBottom": 16,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#868686",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "codeo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#868686",
                "fontFamily": "Raleway-SemiBold",
                "fontSize": 15,
                "lineHeight": 18,
                "marginLeft": 8,
                "textTransform": "capitalize",
              },
            ]
          }
        >
          common.email
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="auth.add_email"
          returnKeyLabel="auth.next"
          returnKeyType="next"
          style={
            Object {
              "flex": 1,
            }
          }
          testID="input_email"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 120,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#9A9A9D",
            "borderBottomWidth": 0.5,
            "marginBottom": 8,
            "paddingBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginBottom": 16,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#868686",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "codeo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#868686",
                "fontFamily": "Raleway-SemiBold",
                "fontSize": 15,
                "lineHeight": 18,
                "marginLeft": 8,
                "textTransform": "capitalize",
              },
            ]
          }
        >
          common.password
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="auth.add_password"
          returnKeyLabel="common.next"
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "flex": 1,
            }
          }
          testID="input_password"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#5956E9",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "codeo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 120,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "borderBottomColor": "#9A9A9D",
            "borderBottomWidth": 0.5,
            "marginBottom": 8,
            "paddingBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginBottom": 16,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#868686",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "codeo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#868686",
                "fontFamily": "Raleway-SemiBold",
                "fontSize": 15,
                "lineHeight": 18,
                "marginLeft": 8,
                "textTransform": "capitalize",
              },
            ]
          }
        >
          auth.repeat_password
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="auth.confirm_password"
          returnKeyType="send"
          secureTextEntry={true}
          style={
            Object {
              "flex": 1,
            }
          }
          testID="input_comfirm_password"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#5956E9",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "codeo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#5956E9",
          "borderRadius": 10,
          "flex": 1,
          "height": 70,
          "justifyContent": "center",
          "maxHeight": 70,
          "opacity": 0.75,
          "width": "100%",
        }
      }
      testID="btn_sign_up"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#868686",
              "fontFamily": "Raleway-Bold",
              "fontSize": 20,
              "lineHeight": 24,
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        auth.sign_up
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn_login"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#5956E9",
              "fontFamily": "Raleway-Regular",
              "fontSize": 16,
              "lineHeight": 18,
              "textAlign": "center",
            },
          ]
        }
      >
        auth.have_account
      </Text>
    </View>
  </View>
</View>
`;
